### DISCUSSION TITLE
How can you manipulate the forum BBCode stuff?

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#199] [Help]
That bar with options for links, bold text and such stuff. I have a problem with it. I really can't find out how to manipulate it. For my current project, the smiley remapper (#197) I looked at that bar since I need to replace `:)` with `:smile:` so I thought I'm just gonna change an event. Well, onmousedown:
```
function() {
  if (button.call) {
    eval(button.call)()
  }
  setTimeout(function() {
    markup(button)
  }, 1);
  return false
}
```
Quite obvious how to manipulate that, right?
No really, I have no clue what I can do. I wasn't able to find out anything about `button.call` and the other stuff.

**Comment by Explosion-Scratch** [comment]
```js
string.replace('find this', 'replace with this');
```

**Comment by mxmou** [comment]
[This](https://github.com/markitup/1.x) is the library used for the editor, if that helps.

**Comment by mxmou** [comment]
If you find a way to ensure your userscript runs before markitup loads you could just change the `mySettings` object.

**Comment by GrahamSH-LLK** [comment]
@JeffaloBob does something like this in his userscript, maybe he can help.
